backend/logic/deploy/records.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * Funcitonality for deployment records in the database
+ */
+const { ShopDeploymentStatuses } = require('../../enums')
+const { ShopDeployment, ShopDeploymentName } = require('../../models')
+const { assert } = require('../../utils/validators')
+const { getLogger } = require('../../utils/logger')
+
+const { DuplicateDeploymentError } = require('./common')
+
+const log = getLogger('logic.deploy.records')
+
+// Max age of a deployment before it is considered as failed.
+const MAX_PENDING_DEPLOYMENT_AGE = 10 * 60 * 1000 // 10 min
+
+/**
+ * Set a deployment as failed with given errorMessage
+ *
+ * @param deployment {object} - ShopDeployment model instance
+ * @param errorMessage {string} - Error message to store
+ */
+async function failDeployment(deployment, errorMessage) {
+  await deployment.update({
+    status: ShopDeploymentStatuses.Failure,
+    error: errorMessage
+  })
+}
+
+/**
+ * Set a deployment as succeeded
+ *
+ * @param deployment {object} - ShopDeployment model instance
+ * @param props {object} - Any extra model propreties to set
+ */
+async function passDeployment(deployment, props) {
+  await deployment.update({
+    status: ShopDeploymentStatuses.Success,
+    ...props
+  })
+}
+
+/**
+ * Create a shop deployment record
+ *
+ * @param shopId {number} - Shop ID to create a deployment for
+ * @param uuid {string} - A UUID identifier that can be used for later retrieval
```

Comment:
Why do we need a uuid? Could we simply use shop_deployments.id?

Reply:
I chose uuid because it makes things a lot cleaner.  A UUID can be created before the deployment process starts without having to fiddle with deployment locking DB records.

Specifically, the deployment DB record is first created in `deploy()` function, which is called in a bull queue job.  The only alternative would be to create the deployment DB record ahead of time, which doesn't really make sense, IMO.  Leaves complexity in the job processor instead of the express handler.

Reply:
Sounds good. I'll leave it up to you to decide.